build.ps1

```diff
@@ -0,0 +1,264 @@
+#requires -version 4
+
+<#
+.SYNOPSIS
+Builds this repository.
+
+.DESCRIPTION
+This build script installs required tools and runs an MSBuild command on this repository.
+This script can be used to invoke various targets, such as targets to produce packages,
+build projects, run tests, and generate code.
+
+.PARAMETER RepoPath
+The folder to build. Defaults to the folder containing this script. This will be removed soon.
+
+.PARAMETER CI
+Sets up CI specific settings and variables.
+
+.PARAMETER Restore
+Run restore on projects.
+
+.PARAMETER Build
+Compile projects.
+
+.PARAMETER Pack
+Produce packages.
+
+.PARAMETER Test
+Run tests.
```

Comment:
Not sure I understood whether this *only* runs tests, or does it also build some subset of things required to run tests? And if it literally only *runs* tests (i.e. it assumes all binaries are already built and in the right location), when would that be needed?

Reply:
`build.cmd -test` will build AND test. We could add a gesture for just running tests without building, but I don't think re-running tests without changes is very common.

Reply:
@halter73 does this all the time, re-running tests in a loop trying to repro flaky issues.

Reply:
OK I was thinking it presumably builds & runs tests.

The case of re-running tests in a loop happens, but is rare, so I didn't think that would be a frequently used case.

Reply:
if the build were incremental then it would work both ways...

Reply:
Builds should be incremental once we finish #4246.

> re-running tests in a loop trying to repro flaky issues.

You can do it with the build.cmd script, it's just not the default. On *nix, `build.sh --no-restore --no-build --test`. On Windows, it's `./build.ps1 -restore:$false -build:$false -test`. You can also use `dotnet test --no-build` on the project you're trying to use.